docs/en/stores/supplemental.md

```diff
@@ -0,0 +1,394 @@
+# StoreProvider
+
+## StoreProvider API
+
+The StoreProvider accepts three properties
+
+-   `renderer`: A render function that has the store injected in order to access state and pass processes to child widgets.
+-   `stateKey`: The key of the state in the registry.
+-   `paths` (optional): A function to connect the Container to sections of the state.
+
+## Invalidation
+
+The `StoreProvider` has two main ways to trigger invalidation and cause a rerender.
+
+1.  The recommended approach is to register `path`s on container creation to ensure invalidation will only occur when state you are interested in changes.
+1.  A catch-all when no `path`s are defined for the container, it will invalidate when _any_ data changes in the store.
+
+## Pre-typing
+
+A pre-typed container can be created by extending the standard `StoreProvider` and passing the `State` type as a generic.
+
+> TypedStoreProvider.ts
+
+```ts
+import { State } from './interface';
+
+export class TypedStoreProvider extends StoreProvider<State> {}
+```
+
+**However** in order for TypeScript to infer this correctly when using w(), the generic will need to be explicitly passed.
```

Comment:
Using `tsx` infers the renderer functions arguments correctly, do you think we should demonstrate this?

Reply:
That sounds great. Why does it work for `tsx` and not `w()`?